staging/src/k8s.io/api/core/v1/types.go

```diff
@@ -5163,6 +5163,9 @@ const (
 	// Local ephemeral storage, in bytes. (500Gi = 500GiB = 500 * 1024 * 1024 * 1024)
 	// The resource name for ResourceEphemeralStorage is alpha and it can change across releases.
 	ResourceEphemeralStorage ResourceName = "ephemeral-storage"
+	// Swap, in bytes. (500Gi = 500GiB = 500 * 1024 * 1024 * 1024)
+	// Supported if the feature gate NodeSwap is enabled
+	ResourceSwap ResourceName = "swap"
```

Comment:
This needs to first land in https://github.com/kubernetes/kubernetes/pull/105271, then we can rebase this PR to be much smaller

Reply:
But this PR https://github.com/kubernetes/kubernetes/pull/105271 has not been merged, we need to wait for it to be merged?

Reply:
It would be better if that one merged first, it will simplify this one once it's rebased.